R/simulate.R

```diff
@@ -1,57 +1,94 @@
-##' Simulate a dust model over time. This is a wrapper around calling
-##' `$run()` and `$state()` repeatedly without doing anything very
-##' interesting with the data. It is provided mostly as a
-##' quick-and-dirty way of getting started with a model.
+##' Simulate a dust model.  This is a helper function that is subject
+##' to change; see Details below.
 ##'
-##' @title Simulate a dust model
+##' This function has an interface that we expect to change once
+##' multi-parameter "dust" objects are fully supported.  For now, it
+##' is designed to be used where we want to simulate a number of
+##' trajectories given a vector of model data and a matrix of initial
+##' state.  For our immediate use case this is for simulating at the
+##' end of an MCMC where want to generate a posterior distribution of
+##' trajectories.
 ##'
-##' @param model A model, compiled with `dust::dust()` and initialised
+##' @section Random number generation:
 ##'
-##' @param steps A vector of steps - the first step must be same as
-##'   the model's step (i.e., `$step()`)
+##' This implementation leaves a number of issues that we will resolve
```

Comment:
The issue is the roundtrip. So we have our particle filter which has n particles each with an RNG state and we want to simulate m trajectories.

So going from filter -> predict we should grab some representation of the RNG state (either the first particle's state or the whole vector and set it into the object doing the simulation. If n < m or if we only take the first particle we'll do a bunch of jumps on the rest - that's fine, but needs work in dust.

Then once we're done with the simulation we need to pull this state back.

I guess what you're suggesting is to do a long_jump on the dust object after copying its state so that it's "moved along" and copy some state over into the bit doing the prediction?

In any case, I think that there are sufficient judgement calls here we should work through this on a separate PR!

Reply:
As I've added it now, see #87